plugins/docker/src/index.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  Auto,
+  determineNextVersion,
+  execPromise,
+  IPlugin,
+  getCurrentBranch,
+  DEFAULT_PRERELEASE_BRANCHES,
+  validatePluginConfiguration,
+} from "@auto-it/core";
+import * as t from "io-ts";
+import { inc, ReleaseType } from "semver";
+
+const pluginOptions = t.intersection([
+  t.interface({
+    /** The target registry to push too */
+    registry: t.string,
+  }),
+  t.partial({
+    /** Whether to tag non-prerelease images with latest */
+    tagLatest: t.boolean,
+    /** The source image, if a static name and/or tag */
+    image: t.string,
+  }),
+]);
+
+export type IDockerPluginOptions = t.TypeOf<typeof pluginOptions>;
+
+/** Manages tagging and publishing to docker registries. */
+export default class DockerPlugin implements IPlugin {
+  /** The name of the plugin */
+  name = "docker";
+
+  /** The calculated docker tags for publish */
+  private calculatedTags?: string[];
+
+  /** Initialize the plugin with it's options */
+  constructor(private readonly options: IDockerPluginOptions) {
+    this.options.image = options.image || process.env.IMAGE;
+  }
+
+  /** Tap into auto plugin points. */
+  apply(auto: Auto) {
+    /** Gets the latest tag for a given branch */
+    async function getTag() {
+      try {
+        return await auto.git!.getLatestTagInBranch();
+      } catch (error) {
+        return auto.prefixRelease("0.0.0");
+      }
+    }
+
+    auto.hooks.validateConfig.tapPromise(this.name, async (name, options) => {
+      if (name === this.name || name === `@auto-it/${this.name}`) {
+        if (Array.isArray(options)) {
+          const errors = await Promise.all(
+            options.map((o) =>
+              validatePluginConfiguration(this.name, pluginOptions, o)
+            )
+          );
+
+          return errors.reduce((acc, item) => [...acc, ...item], []);
+        }
```

Comment:
I don't think this if statement is necessary. This would validate if a user provided an array of this plugin's option. I don't think this plugin allows for that

Reply:
Good catch, got copy pasta'd. Tempted to add this later, but not necessary now.